### PR TITLE
feat: Fix eval metrics, perdi bugs, CSV filter, download models data

### PR DESCRIPTION
**Pull request by Ciroye**

### CODE REVIEW

File: backend/app/domain/services/base/model.py [hunk 1]
```diff
@@ -220,6 +221,10 @@ def single_model_prediction_submit(
             )
             external_validator_url = external_validator.get("url", None)
             external_validator_artifacts = external_validator.get("artifacts", None)
+        else:
+            amount_require_examples = -1
```
Comment (remg1997) [inline backend/app/domain/services/base/model.py:225]: total_required_examples would be the correct naming

File: backend/app/domain/services/base/model.py [hunk 2]
```diff
@@ -450,3 +456,27 @@ def update_model_info(
         return self.model_repository.update_model_info(
             model_id, name, desc, longdesc, params, languages, license, source_url
         )
+
+    def download_results_models(self, task_id: int):
```
Comment (remg1997) [inline backend/app/domain/services/base/model.py:460]: download_model_results

File: backend/app/domain/services/base/task.py [hunk 3]
```diff
@@ -142,6 +142,7 @@ def get_task_with_round_info_by_task_id(self, task_id: int):
         task_and_round_info = self.task_repository.get_task_with_round_info_by_task_id(
             task_id
         )
+        print(task_and_round_info)
```
Comment (remg1997) [inline backend/app/domain/services/base/task.py:145]: Any reason to leave this print?

**Review by remg1997 — CHANGES_REQUESTED** [review]
Minor changes requested

File: backend/app/domain/services/base/task.py (hunk 3 above)
Comment (Ciroye) [reply backend/app/domain/services/base/task.py:145]: No 🤡

**Review by remg1997 — APPROVED** [review]

**Pull request merged**